### ISSUE TITLE
OC10 - Class 'OC\\\\MemCache\\\\ArrayCache' not found

### ISSUE THREAD

**Issue by mdusher** [owncloud/core#29212]
### Steps to reproduce
1. Performed upgrade from 8.2.11 to 10.0.3
2. Attempted to login
3. Received exception

### Expected behaviour
Login with no exceptions

### Actual behaviour
Login and received exception

### Server configuration
**Operating system**: Redhat 6

**Web server:** Apache 2.2.15

**Database:** MariaDB 10.0.31

**PHP version:** 7.0.22

**ownCloud version:** 10.0.3

**Updated from an older ownCloud or fresh install:** Updated from 8.2.11

**Where did you install ownCloud from:** https://owncloud.org/install/#edition

**Signing status (ownCloud 9.0 and above):** 
```
Integrity checker has been disabled. Integrity cannot be verified.
```

**The content of config/config.php:**

```
{
    "system": {
        "debug": true,
        "instanceid": "50c824ae04bf7",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloudstor-uat.aarnet.edu.au"
        ],
        "datadirectory": "\/cloudstor\/data\/owncloud\/data",
        "version": "10.0.3.3",
        "dbtype": "mysql",
        "dbname": "cloudstor1003",
        "dbhost": "127.0.0.1:16033",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "",
        "installed": true,
        "default_language": "en_GB",
        "defaultapp": "files",
        "knowledgebaseenabled": true,
        "enable_avatars": false,
        "allow_user_to_change_display_name": false,
        "session_lifetime": 86400,
        "session_keepalive": true,
        "token_auth_enforced": false,
        "mail_domain": "aarnet.edu.au",
        "mail_from_address": "cloudstor-noreply",
        "mail_smtpmode": "php",
        "overwriteprotocol": "https",
        "overwrite.cli.url": "https:\/\/cloudstor-uat.aarnet.edu.au\/10\/",
        "htaccess.RewriteBase": "\/10",
        "trashbin_retention_obligation": "30, auto",
        "appcodechecker": false,
        "updatechecker": false,
        "has_internet_connection": true,
        "check_for_working_webdav": false,
        "check_for_working_htaccess": true,
        "log_type": "owncloud",
        "logfile": "\/cloudstor\/logs\/owncloud-10.0.3\/owncloud.log",
        "loglevel": 0,
        "logtimezone": "UTC",
        "log_query": false,
        "cron_log": true,
        "log_rotate_size": 104857600,
        "customclient_desktop": "https:\/\/cloudstor.aarnet.edu.au\/client-download\/",
        "customclient_android": "https:\/\/play.google.com\/store\/apps\/details?id=au.edu.aarnet.cloudstor.android",
        "customclient_ios": "https:\/\/itunes.apple.com\/au\/app\/cloudstor\/id1215476371?mt=8",
        "appstore.experimental.enabled": false,
        "apps_paths": [
            {
                "path": "\/cloudstor\/www\/owncloud-10.0.3\/apps",
                "url": "\/apps",
                "writable": true
            },
            {
                "path": "\/cloudstor\/www\/owncloud-10.0.3\/3rdparty-apps",
                "url": "\/3rdparty-apps",
                "writable": true
            }
        ],
        "enable_previews": true,
        "enabledPreviewProviders": [
            "OC\\Preview\\PNG",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\GIF",
            "OC\\Preview\\BMP",
            "OC\\Preview\\XBitmap",
            "OC\\Preview\\MP3",
            "OC\\Preview\\TXT",
            "OC\\Preview\\MarkDown"
        ],
        "maintenance": false,
        "singleuser": false,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "127.0.0.1",
            "port": 6379,
            "timeout": 0
        },
        "memcached_servers": [
            [
                "localhost",
                11211
            ]
        ],
        "cache_path": "\/cloudstor\/data\/cache",
        "blacklisted_files": [
            ".htaccess"
        ],
        "share_folder": "\/Shared",
        "cipher": "AES-256-CFB",
        "minimum.supported.desktop.version": "2.2.2",
        "quota_include_external_storage": false,
        "filesystem_check_changes": 1,
        "asset-pipeline.enabled": false,
        "filesystem_cache_readonly": false,
        "forwarded_for_headers": [
            "HTTP_X_FORWARDED",
            "HTTP_FORWARDED_FOR"
        ],
        "filelocking.enabled": true,
        "filelocking.ttl": 3600,
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "upgrade.disable-web": true,
        "upgrade.automatic-app-update": true,
        "integrity.check.disabled": true
    }
}

```

**List of activated apps:**

```
  - activity: 2.3.4
  - cloudstortheme: 1.0.0
  - collections: 1.0.0
  - comments: 0.3.0
  - configreport: 0.1.1
  - dav: 0.3.0
  - direct_menu: 0.1.0
  - federatedfilesharing: 0.3.1
  - federation: 0.1.0
  - files: 1.5.1
  - files_clipboard: 0.6.4
  - files_external: 0.7.1
  - files_pdfviewer: 0.8.2
  - files_sharing: 0.10.1
  - files_texteditor: 2.2.1
  - files_trashbin: 0.9.1
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - impersonate: 0.1.0
  - notifications: 0.3.1
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - templateeditor: 0.1
  - updatenotification: 0.2.1
  - user_saml: 0.4
```

**Are you using external storage, if yes which one:** No.

**Are you using encryption:** No.

**Are you using an external user-backend, if yes which one:** No, unless you count user_saml.

### Client configuration
**Browser:** Chrome 60.0.3112.113 

**Operating system:** Windows 10

#### Web server error log
```
No relevant logs
```

#### ownCloud log (data/owncloud.log)
```
{
  "reqId": "1dxzZc3BMvU7BhQ6PL1w",
  "level": 3,
  "time": "2017-10-12T00:09:10+00:00",
  "remoteAddr": "10.42.4.110",
  "user": "Michael",
  "app": "index",
  "method": "GET",
  "url": "/10/login",
  "message": "Exception: {\"Exception\":\"Error\",\"Message\":\"Class 'OC\\\\MemCache\\\\ArrayCache' not found\",\"Code\":0,\"Trace\":\"#0 \\/cloudstor\\/www\\/owncloud-10\\/apps\\/files_sharing\\/lib\\/External\\/Storage.php(92): OC\\\\Files\\\\Storage\\\\DAV->__construct(Array)\\n#1 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/Files\\/Storage\\/StorageFactory.php(82): OCA\\\\Files_Sharing\\\\External\\\\Storage->__construct(Array)\\n#2 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/Files\\/Mount\\/MountPoint.php(138): OC\\\\Files\\\\Storage\\\\StorageFactory->getInstance(Object(OCA\\\\Files_Sharing\\\\External\\\\Mount), '\\\\\\\\OCA\\\\\\\\Files_Shar...', Array)\\n#3 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/Files\\/Mount\\/MountPoint.php(161): OC\\\\Files\\\\Mount\\\\MountPoint->createStorage()\\n#4 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/Files\\/Mount\\/MountPoint.php(251): OC\\\\Files\\\\Mount\\\\MountPoint->getStorage()\\n#5 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/Files\\/Config\\/UserMountCache.php(92): OC\\\\Files\\\\Mount\\\\MountPoint->getStorageRootId()\\n#6 [internal function]: OC\\\\Files\\\\Config\\\\UserMountCache->OC\\\\Files\\\\Config\\\\{closure}(Object(OCA\\\\Files_Sharing\\\\External\\\\Mount))\\n#7 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/Files\\/Config\\/UserMountCache.php(97): array_map(Object(Closure), Array)\\n#8 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/Files\\/Config\\/MountProviderCollection.php(134): OC\\\\Files\\\\Config\\\\UserMountCache->registerMounts(Object(OC\\\\User\\\\User), Array)\\n#9 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/Files\\/Filesystem.php(448): OC\\\\Files\\\\Config\\\\MountProviderCollection->registerMounts(Object(OC\\\\User\\\\User), Array)\\n#10 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/Files\\/Filesystem.php(375): OC\\\\Files\\\\Filesystem::initMountPoints('Michael.Usher@a...')\\n#11 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/legacy\\/util.php(297): OC\\\\Files\\\\Filesystem::init('Michael.Usher@a...', '\\/Michael.Usher@...')\\n#12 \\/cloudstor\\/www\\/owncloud-10\\/3rdparty-apps\\/user_saml\\/lib\\/hooks.php(154): OC_Util::setupFS('Michael.Usher@a...')\\n#13 \\/cloudstor\\/www\\/owncloud-10\\/3rdparty-apps\\/user_saml\\/lib\\/hooks.php(54): OCA\\\\User_Saml\\\\Hooks::updateUserData('Michael.Usher@a...', Array)\\n#14 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/legacy\\/hook.php(105): OCA\\\\User_Saml\\\\Hooks::postLogin(Array)\\n#15 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/Server.php(318): OC_Hook::emit('OC_User', 'post_login', Array)\\n#16 [internal function]: OC\\\\Server->OC\\\\{closure}(Object(OC\\\\User\\\\User), '')\\n#17 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/Hooks\\/EmitterTrait.php(99): call_user_func_array(Object(Closure), Array)\\n#18 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/Hooks\\/PublicEmitter.php(33): OC\\\\Hooks\\\\BasicEmitter->emit('\\\\\\\\OC\\\\\\\\User', 'postLogin', Array)\\n#19 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/User\\/Session.php(477): OC\\\\Hooks\\\\PublicEmitter->emit('\\\\\\\\OC\\\\\\\\User', 'postLogin', Array)\\n#20 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/User\\/Session.php(301): OC\\\\User\\\\Session->loginWithPassword(*** sensitive parameters replaced ***)\\n#21 \\/cloudstor\\/www\\/owncloud-10\\/3rdparty-apps\\/user_saml\\/lib\\/util.php(29): OC\\\\User\\\\Session->login(*** sensitive parameters replaced ***)\\n#22 \\/cloudstor\\/www\\/owncloud-10\\/3rdparty-apps\\/user_saml\\/appinfo\\/app.php(43): OCA\\\\User_Saml\\\\Util::doLogin('', '')\\n#23 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/legacy\\/app.php(213): require_once('\\/cloudstor\\/www\\/...')\\n#24 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/legacy\\/app.php(164): OC_App::requireAppFile('user_saml')\\n#25 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/legacy\\/app.php(121): OC_App::loadApp('user_saml')\\n#26 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/legacy\\/user.php(128): OC_App::loadApps(Array)\\n#27 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/base.php(655): OC_User::setupBackends()\\n#28 \\/cloudstor\\/www\\/owncloud-10\\/lib\\/base.php(1013): OC::init()\\n#29 \\/cloudstor\\/www\\/owncloud-10\\/index.php(54): require_once('\\/cloudstor\\/www\\/...')\\n#30 {main}\",\"File\":\"\\/cloudstor\\/www\\/owncloud-10\\/lib\\/private\\/Files\\/Storage\\/DAV.php\",\"Line\":90}"
}
```

**Comment by mdusher** [comment]
Found the cause of this and have created a pull request for it in #29213

**Comment by DeepDiver1975** [comment]
fixed with #29213
